src/main/scala/variant/Atomic.scala

```diff
@@ -76,22 +77,21 @@ case object Atomic extends Variant(
   override def addVariantEffect(move: Move): Move = explodeSurroundingPieces(move)
 
   /**
-   * Since a king may walk into the path of another king, it is more difficult to win when your opponent only has a
-   * king left.
+   * Since kings cannot confine each other, if either player has only a king
+   * then either a queen or multiple pieces are required for checkmate.
    */
   private def insufficientAtomicWinningMaterial(board: Board) = {
-    val whiteActors = board.actorsOf(White)
-    val blackActors = board.actorsOf(Black)
-    val allActors = board.actors
-    lazy val allPieces = board.actors.values.map(_.piece).filter(_ isNot King)
-
-    // One player must only have their king left
-    if (whiteActors.size != 1 && blackActors.size != 1) false
-    else {
-      // You can mate with a queen or a pawn that is later promoted a a queen, but not with just one of any other piece
-      // or with just a king
-      allPieces.size == 1 && !allPieces.exists(p => p.is(Queen) || p.is(Pawn)) || allActors.size == 2
-    }
+    val bishopsOnly = board.pieces forall { p => (p._2 is King) || (p._2 is Bishop) }
+    lazy val bishopsOnOppositeColors = InsufficientMatingMaterial.bishopsOnOppositeColors(board)
+    lazy val minorsOnly = board.pieces forall { p => (p._2 is King) || (p._2 is Rook) || (p._2 is Bishop) || (p._2 is Knight) }
+    lazy val rookExists = board.pieces exists { _._2 is Rook }
```

Comment:
I didn't want to call `board.count(Rook)` but unfortunately there is no convenience function `board.exists(Rook)`.